File: testcases/vertex_ai/GeometryTest_7836f66d.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class GeometryTest {

    @Test
    void calculateCircleArea_positiveRadius_returnsCorrectArea() {
        Geometry geometry = new Geometry();
        double radius = 5.0;
        double expectedArea = Math.PI * radius * radius;
        assertEquals(expectedArea, geometry.calculateCircleArea(radius), 0.001);
    }

    @Test
    void calculateCircleArea_zeroRadius_returnsZeroArea() {
        Geometry geometry = new Geometry();
        double radius = 0.0;
        double expectedArea = 0.0;
        assertEquals(expectedArea, geometry.calculateCircleArea(radius), 0.001);
    }

    @Test
    void calculateCircleArea_largeRadius_returnsCorrectArea() {
        Geometry geometry = new Geometry();
        double radius = 100.0;
        double expectedArea = Math.PI * radius * radius;
        assertEquals(expectedArea, geometry.calculateCircleArea(radius), 0.001);
    }

    @Test
    void calculateCircleArea_negativeRadius_throwsIllegalArgumentException() {
        Geometry geometry = new Geometry();
        double radius = -5.0;
        assertThrows(IllegalArgumentException.class, () -> geometry.calculateCircleArea(radius));
    }

    @Test
    void calculateCircleArea_smallPositiveRadius_returnsCorrectArea() {
        Geometry geometry = new Geometry();
        double radius = 0.1;
        double expectedArea = Math.PI * radius * radius;
        assertEquals(expectedArea, geometry.calculateCircleArea(radius), 0.001);
    }

    @Test
    void calculateRectangleArea_positiveLengthWidth_returnsCorrectArea() {
        Geometry geometry = new Geometry();
        double length = 5.0;
        double width = 10.0;
        double expectedArea = length * width;
        assertEquals(expectedArea, geometry.calculateRectangleArea(length, width), 0.001);
    }

    @Test
    void calculateRectangleArea_zeroLengthWidth_returnsZeroArea() {
        Geometry geometry = new Geometry();
        double length = 0.0;
        double width = 0.0;
        double expectedArea = 0.0;
        assertEquals(expectedArea, geometry.calculateRectangleArea(length, width), 0.001);
    }

    @Test
    void calculateRectangleArea_zeroLengthPositiveWidth_returnsZeroArea() {
        Geometry geometry = new Geometry();
        double length = 0.0;
        double width = 10.0;
        double expectedArea = 0.0;
        assertEquals(expectedArea, geometry.calculateRectangleArea(length, width), 0.001);
    }

    @Test
    void calculateRectangleArea_positiveLengthZeroWidth_returnsZeroArea() {
        Geometry geometry = new Geometry();
        double length = 5.0;
        double width = 0.0;
        double expectedArea = 0.0;
        assertEquals(expectedArea, geometry.calculateRectangleArea(length, width), 0.001);
    }

    @Test
    void calculateRectangleArea_negativeLengthPositiveWidth_throwsIllegalArgumentException() {
        Geometry geometry = new Geometry();
        double length = -5.0;
        double width = 10.0;
        assertThrows(IllegalArgumentException.class, () -> geometry.calculateRectangleArea(length, width));
    }

    @Test
    void calculateRectangleArea_positiveLengthNegativeWidth_throwsIllegalArgumentException() {
        Geometry geometry = new Geometry();
        double length = 5.0;
        double width = -10.0;
        assertThrows(IllegalArgumentException.class, () -> geometry.calculateRectangleArea(length, width));
    }

    @Test
    void calculateRectangleArea_negativeLengthNegativeWidth_throwsIllegalArgumentException() {
        Geometry geometry = new Geometry();
        double length = -5.0;
        double width = -10.0;
        assertThrows(IllegalArgumentException.class, () -> geometry.calculateRectangleArea(length, width));
    }

    @Test
    void calculateTriangleArea_positiveBaseHeight_returnsCorrectArea() {
        Geometry geometry = new Geometry();
        double base = 5.0;
        double height = 10.0;
        double expectedArea = 0.5 * base * height;
        assertEquals(expectedArea, geometry.calculateTriangleArea(base, height), 0.001);
    }

    @Test
    void calculateTriangleArea_zeroBaseHeight_returnsZeroArea() {
        Geometry geometry = new Geometry();
        double base = 0.0;
        double height = 0.0;
        double expectedArea = 0.0;
        assertEquals(expectedArea, geometry.calculateTriangleArea(base, height), 0.001);
    }

    @Test
    void calculateTriangleArea_zeroBasePositiveHeight_returnsZeroArea() {
        Geometry geometry = new Geometry();
        double base = 0.0;
        double height = 10.0;
        double expectedArea = 0.0;
        assertEquals(expectedArea, geometry.calculateTriangleArea(base, height), 0.001);
    }

    @Test
    void calculateTriangleArea_positiveBaseZeroHeight_returnsZeroArea() {
        Geometry geometry = new Geometry();
        double base = 5.0;
        double height = 0.0;
        double expectedArea = 0.0;
        assertEquals(expectedArea, geometry.calculateTriangleArea(base, height), 0.001);
    }

    @Test
    void calculateTriangleArea_negativeBasePositiveHeight_throwsIllegalArgumentException() {
        Geometry geometry = new Geometry();
        double base = -5.0;
        double height = 10.0;
        assertThrows(IllegalArgumentException.class, () -> geometry.calculateTriangleArea(base, height));
    }

    @Test
    void calculateTriangleArea_positiveBaseNegativeHeight_throwsIllegalArgumentException() {
        Geometry geometry = new Geometry();
        double base = 5.0;
        double height = -10.0;
        assertThrows(IllegalArgumentException.class, () -> geometry.calculateTriangleArea(base, height));
    }

    @Test
    void calculateTriangleArea_negativeBaseNegativeHeight_throwsIllegalArgumentException() {
        Geometry geometry = new Geometry();
        double base = -5.0;
        double height = -10.0;
        assertThrows(IllegalArgumentException.class, () -> geometry.calculateTriangleArea(base, height));
    }

    @Test
    void calculateCirclePerimeter_positiveRadius_returnsCorrectPerimeter() {
        Geometry geometry = new Geometry();
        double radius = 5.0;
        double expectedPerimeter = 2 * Math.PI * radius;
        assertEquals(expectedPerimeter, geometry.calculateCirclePerimeter(radius), 0.001);
    }

    @Test
    void calculateCirclePerimeter_zeroRadius_returnsZeroPerimeter() {
        Geometry geometry = new Geometry();
        double radius = 0.0;
        double expectedPerimeter = 0.0;
        assertEquals(expectedPerimeter, geometry.calculateCirclePerimeter(radius), 0.001);
    }

    @Test
    void calculateCirclePerimeter_largeRadius_returnsCorrectPerimeter() {
        Geometry geometry = new Geometry();
        double radius = 100.0;
        double expectedPerimeter = 2 * Math.PI * radius;
        assertEquals(expectedPerimeter, geometry.calculateCirclePerimeter(radius), 0.001);
    }

    @Test
    void calculateCirclePerimeter_negativeRadius_throwsIllegalArgumentException() {
        Geometry geometry = new Geometry();
        double radius = -5.0;
        assertThrows(IllegalArgumentException.class, () -> geometry.calculateCirclePerimeter(radius));
    }

    @Test
    void calculateCirclePerimeter_smallPositiveRadius_returnsCorrectPerimeter() {
        Geometry geometry = new Geometry();
        double radius = 0.1;
        double expectedPerimeter = 2 * Math.PI * radius;
        assertEquals(expectedPerimeter, geometry.calculateCirclePerimeter(radius), 0.001);
    }

    @Test
    void calculateRectanglePerimeter_positiveLengthWidth_returnsCorrectPerimeter() {
        Geometry geometry = new Geometry();
        double length = 5.0;
        double width = 10.0;
        double expectedPerimeter = 2 * (length + width);
        assertEquals(expectedPerimeter, geometry.calculateRectanglePerimeter(length, width), 0.001);
    }

    @Test
    void calculateRectanglePerimeter_zeroLengthWidth_returnsZeroPerimeter() {
        Geometry geometry = new Geometry();
        double length = 0.0;
        double width = 0.0;
        double expectedPerimeter = 0.0;
        assertEquals(expectedPerimeter, geometry.calculateRectanglePerimeter(length, width), 0.001);
    }

    @Test
    void calculateRectanglePerimeter_zeroLengthPositiveWidth_returnsCorrectPerimeter() {
        Geometry geometry = new Geometry();
        double length = 0.0;
        double width = 10.0;
        double expectedPerimeter = 2 * (length + width);
        assertEquals(expectedPerimeter, geometry.calculateRectanglePerimeter(length, width), 0.001);
    }

    @Test
    void calculateRectanglePerimeter_positiveLengthZeroWidth_returnsCorrectPerimeter() {
        Geometry geometry = new Geometry();
        double length = 5.0;
        double width = 0.0;
        double expectedPerimeter = 2 * (length + width);
        assertEquals(expectedPerimeter, geometry.calculateRectanglePerimeter(length, width), 0.001);
    }

    @Test
    void calculateRectanglePerimeter_negativeLengthPositiveWidth_throwsIllegalArgumentException() {
        Geometry geometry = new Geometry();
        double length = -5.0;
        double width = 10.0;
        assertThrows(IllegalArgumentException.class, () -> geometry.calculateRectanglePerimeter(length, width));
    }

    @Test
    void calculateRectanglePerimeter_positiveLengthNegativeWidth_throwsIllegalArgumentException() {
        Geometry geometry = new Geometry();
        double length = 5.0;
        double width = -10.0;
        assertThrows(IllegalArgumentException.class, () -> geometry.calculateRectanglePerimeter(length, width));
    }

    @Test
    void calculateRectanglePerimeter_negativeLengthNegativeWidth_throwsIllegalArgumentException() {
        Geometry geometry = new Geometry();
        double length = -5.0;
        double width = -10.0;
        assertThrows(IllegalArgumentException.class, () -> geometry.calculateRectanglePerimeter(length, width));
    }

    @Test
    void calculateDistance_positiveCoordinates_returnsCorrectDistance() {
        Geometry geometry = new Geometry();
        double x1 = 0.0;
        double y1 = 0.0;
        double x2 = 3.0;
        double y2 = 4.0;
        double expectedDistance = 5.0;
        assertEquals(expectedDistance, geometry.calculateDistance(x1, y1, x2, y2), 0.001);
    }

    @Test
    void calculateDistance_negativeCoordinates_returnsCorrectDistance() {
        Geometry geometry = new Geometry();
        double x1 = -1.0;
        double y1 = -2.0;
        double x2 = 2.0;
        double y2 = 2.0;
        double expectedDistance = 5.0;
        assertEquals(expectedDistance, geometry.calculateDistance(x1, y1, x2, y2), 0.001);
    }

    @Test
    void calculateDistance_sameCoordinates_returnsZeroDistance() {
        Geometry geometry = new Geometry();
        double x1 = 5.0;
        double y1 = 5.0;
        double x2 = 5.0;
        double y2 = 5.0;
        double expectedDistance = 0.0;
        assertEquals(expectedDistance, geometry.calculateDistance(x1, y1, x2, y2), 0.001);
    }

    @Test
    void calculateDistance_zeroCoordinates_returnsCorrectDistance() {
        Geometry geometry = new Geometry();
        double x1 = 0.0;
        double y1 = 0.0;
        double x2 = 0.0;
        double y2 = 0.0;
        double expectedDistance = 0.0;
        assertEquals(expectedDistance, geometry.calculateDistance(x1, y1, x2, y2), 0.001);
    }

    @Test
    void calculateDistance_mixedCoordinates_returnsCorrectDistance() {
        Geometry geometry = new Geometry();
        double x1 = -2.0;
        double y1 = 0.0;
        double x2 = 2.0;
        double y2 = 0.0;
        double expectedDistance = 4.0;
        assertEquals(expectedDistance, geometry.calculateDistance(x1, y1, x2, y2), 0.001);
    }
}

File: testcases/vertex_ai/geometry.test_7836f66d.js
const {
    calculateCircleArea,
    calculateRectangleArea,
    calculateTriangleArea,
    calculateCirclePerimeter,
    calculateRectanglePerimeter,
    calculateDistance
} = require('../../geometry');

describe('calculateCircleArea', () => {
    /** */
     // Test case for valid radius
     // /
    test('should calculate area with valid radius', () => {
        expect(calculateCircleArea(5)).toBeCloseTo(Math.PI * 5 * 5);
    });

    /** */
     // Test case for radius zero
     // /
    test('should return 0 when radius is 0', () => {
        expect(calculateCircleArea(0)).toBe(0);
    });

    /** */
     // Test case for type error when radius is not a number
     // /
    test('should throw TypeError when radius is not a number', () => {
        expect(() => calculateCircleArea('abc')).toThrow(TypeError);
        expect(() => calculateCircleArea(null)).toThrow(TypeError);
        expect(() => calculateCircleArea(undefined)).toThrow(TypeError);
        expect(() => calculateCircleArea({})).toThrow(TypeError);
        expect(() => calculateCircleArea([])).toThrow(TypeError);
    });

    /** */
     // Test case for error when radius is negative
     // /
    test('should throw Error when radius is negative', () => {
        expect(() => calculateCircleArea(-1)).toThrow(Error);
    });

    /** */
     // Test case for a large radius
     // /
    test('should calculate area with a large radius', () => {
        expect(calculateCircleArea(1000)).toBeCloseTo(Math.PI * 1000 * 1000);
    });

    /** */
     // Test case for a decimal radius
     // /
    test('should calculate area with a decimal radius', () => {
        expect(calculateCircleArea(2.5)).toBeCloseTo(Math.PI * 2.5 * 2.5);
    });

    /** */
     // Test case for NaN radius
     // /
    test('should throw TypeError when radius is NaN', () => {
        expect(() => calculateCircleArea(NaN)).toThrow(TypeError);
    });
});

describe('calculateRectangleArea', () => {
    /** */
     // Test case for valid length and width
     // /
    test('should calculate area with valid length and width', () => {
        expect(calculateRectangleArea(5, 10)).toBe(50);
    });

    /** */
     // Test case for zero length
     // /
    test('should return 0 when length is 0', () => {
        expect(calculateRectangleArea(0, 10)).toBe(0);
    });

    /** */
     // Test case for zero width
     // /
    test('should return 0 when width is 0', () => {
        expect(calculateRectangleArea(5, 0)).toBe(0);
    });

    /** */
     // Test case for type error when length is not a number
     // /
    test('should throw TypeError when length is not a number', () => {
        expect(() => calculateRectangleArea('abc', 10)).toThrow(TypeError);
        expect(() => calculateRectangleArea(null, 10)).toThrow(TypeError);
        expect(() => calculateRectangleArea(undefined, 10)).toThrow(TypeError);
    });

    /** */
     // Test case for type error when width is not a number
     // /
    test('should throw TypeError when width is not a number', () => {
        expect(() => calculateRectangleArea(5, 'abc')).toThrow(TypeError);
        expect(() => calculateRectangleArea(5, null)).toThrow(TypeError);
        expect(() => calculateRectangleArea(5, undefined)).toThrow(TypeError);
    });

    /** */
     // Test case for error when length is negative
     // /
    test('should throw Error when length is negative', () => {
        expect(() => calculateRectangleArea(-1, 10)).toThrow(Error);
    });

    /** */
     // Test case for error when width is negative
     // /
    test('should throw Error when width is negative', () => {
        expect(() => calculateRectangleArea(5, -1)).toThrow(Error);
    });

    /** */
     // Test case for large length and width
     // /
    test('should calculate area with large length and width', () => {
        expect(calculateRectangleArea(1000, 2000)).toBe(2000000);
    });

    /** */
     // Test case for decimal length and width
     // /
    test('should calculate area with decimal length and width', () => {
        expect(calculateRectangleArea(2.5, 3.5)).toBe(8.75);
    });

    /** */
     // Test case for NaN length or width
     // /
    test('should throw TypeError when length or width is NaN', () => {
        expect(() => calculateRectangleArea(NaN, 5)).toThrow(TypeError);
        expect(() => calculateRectangleArea(5, NaN)).toThrow(TypeError);
    });
});

describe('calculateTriangleArea', () => {
    /** */
     // Test case for valid base and height
     // /
    test('should calculate area with valid base and height', () => {
        expect(calculateTriangleArea(5, 10)).toBe(25);
    });

    /** */
     // Test case for zero base
     // /
    test('should return 0 when base is 0', () => {
        expect(calculateTriangleArea(0, 10)).toBe(0);
    });

    /** */
     // Test case for zero height
     // /
    test('should return 0 when height is 0', () => {
        expect(calculateTriangleArea(5, 0)).toBe(0);
    });

    /** */
     // Test case for type error when base is not a number
     // /
    test('should throw TypeError when base is not a number', () => {
        expect(() => calculateTriangleArea('abc', 10)).toThrow(TypeError);
        expect(() => calculateTriangleArea(null, 10)).toThrow(TypeError);
        expect(() => calculateTriangleArea(undefined, 10)).toThrow(TypeError);
    });

    /** */
     // Test case for type error when height is not a number
     // /
    test('should throw TypeError when height is not a number', () => {
        expect(() => calculateTriangleArea(5, 'abc')).toThrow(TypeError);
        expect(() => calculateTriangleArea(5, null)).toThrow(TypeError);
        expect(() => calculateTriangleArea(5, undefined)).toThrow(TypeError);
    });

    /** */
     // Test case for error when base is negative
     // /
    test('should throw Error when base is negative', () => {
        expect(() => calculateTriangleArea(-1, 10)).toThrow(Error);
    });

    /** */
     // Test case for error when height is negative
     // /
    test('should throw Error when height is negative', () => {
        expect(() => calculateTriangleArea(5, -1)).toThrow(Error);
    });

    /** */
     // Test case for large base and height
     // /
    test('should calculate area with large base and height', () => {
        expect(calculateTriangleArea(1000, 2000)).toBe(1000000);
    });

    /** */
     // Test case for decimal base and height
     // /
    test('should calculate area with decimal base and height', () => {
        expect(calculateTriangleArea(2.5, 3.5)).toBe(4.375);
    });

    /** */
     // Test case for NaN base or height
     // /
    test('should throw TypeError when base or height is NaN', () => {
        expect(() => calculateTriangleArea(NaN, 5)).toThrow(TypeError);
        expect(() => calculateTriangleArea(5, NaN)).toThrow(TypeError);
    });
});

describe('calculateCirclePerimeter', () => {
    /** */
     // Test case for valid radius
     // /
    test('should calculate perimeter with valid radius', () => {
        expect(calculateCirclePerimeter(5)).toBeCloseTo(2 * Math.PI * 5);
    });

    /** */
     // Test case for radius zero
     // /
    test('should return 0 when radius is 0', () => {
        expect(calculateCirclePerimeter(0)).toBe(0);
    });

    /** */
     // Test case for type error when radius is not a number
     // /
    test('should throw TypeError when radius is not a number', () => {
        expect(() => calculateCirclePerimeter('abc')).toThrow(TypeError);
        expect(() => calculateCirclePerimeter(null)).toThrow(TypeError);
        expect(() => calculateCirclePerimeter(undefined)).toThrow(TypeError);
        expect(() => calculateCirclePerimeter({})).toThrow(TypeError);
        expect(() => calculateCirclePerimeter([])).toThrow(TypeError);
    });

    /** */
     // Test case for error when radius is negative
     // /
    test('should throw Error when radius is negative', () => {
        expect(() => calculateCirclePerimeter(-1)).toThrow(Error);
    });

    /** */
     // Test case for a large radius
     // /
    test('should calculate perimeter with a large radius', () => {
        expect(calculateCirclePerimeter(1000)).toBeCloseTo(2 * Math.PI * 1000);
    });

    /** */
     // Test case for a decimal radius
     // /
    test('should calculate perimeter with a decimal radius', () => {
        expect(calculateCirclePerimeter(2.5)).toBeCloseTo(2 * Math.PI * 2.5);
    });

    /** */
     // Test case for NaN radius
     // /
    test('should throw TypeError when radius is NaN', () => {
        expect(() => calculateCirclePerimeter(NaN)).toThrow(TypeError);
    });
});

describe('calculateRectanglePerimeter', () => {
    /** */
     // Test case for valid length and width
     // /
    test('should calculate perimeter with valid length and width', () => {
        expect(calculateRectanglePerimeter(5, 10)).toBe(30);
    });

    /** */
     // Test case for zero length
     // /
    test('should return correct perimeter when length is 0', () => {
        expect(calculateRectanglePerimeter(0, 10)).toBe(20);
    });

    /** */
     // Test case for zero width
     // /
    test('should return correct perimeter when width is 0', () => {
        expect(calculateRectanglePerimeter(5, 0)).toBe(10);
    });

    /** */
     // Test case for type error when length is not a number
     // /
    test('should throw TypeError when length is not a number', () => {
        expect(() => calculateRectanglePerimeter('abc', 10)).toThrow(TypeError);
        expect(() => calculateRectanglePerimeter(null, 10)).toThrow(TypeError);
        expect(() => calculateRectanglePerimeter(undefined, 10)).toThrow(TypeError);
    });

    /** */
     // Test case for type error when width is not a number
     // /
    test('should throw TypeError when width is not a number', () => {
        expect(() => calculateRectanglePerimeter(5, 'abc')).toThrow(TypeError);
        expect(() => calculateRectanglePerimeter(5, null)).toThrow(TypeError);
        expect(() => calculateRectanglePerimeter(5, undefined)).toThrow(TypeError);
    });

    /** */
     // Test case for error when length is negative
     // /
    test('should throw Error when length is negative', () => {
        expect(() => calculateRectanglePerimeter(-1, 10)).toThrow(Error);
    });

    /** */
     // Test case for error when width is negative
     // /
    test('should throw Error when width is negative', () => {
        expect(() => calculateRectanglePerimeter(5, -1)).toThrow(Error);
    });

    /** */
     // Test case for large length and width
     // /
    test('should calculate perimeter with large length and width', () => {
        expect(calculateRectanglePerimeter(1000, 2000)).toBe(6000);
    });

    /** */
     // Test case for decimal length and width
     // /
    test('should calculate perimeter with decimal length and width', () => {
        expect(calculateRectanglePerimeter(2.5, 3.5)).toBe(12);
    });

     /** */
     // Test case for NaN length or width
     // /
    test('should throw TypeError when length or width is NaN', () => {
        expect(() => calculateRectanglePerimeter(NaN, 5)).toThrow(TypeError);
        expect(() => calculateRectanglePerimeter(5, NaN)).toThrow(TypeError);
    });
});

describe('calculateDistance', () => {
    /** */
     // Test case for valid coordinates
     // /
    test('should calculate distance with valid coordinates', () => {
        expect(calculateDistance(0, 0, 3, 4)).toBe(5);
    });

    /** */
     // Test case for same coordinates
     // /
    test('should return 0 when coordinates are the same', () => {
        expect(calculateDistance(0, 0, 0, 0)).toBe(0);
    });

    /** */
     // Test case for type error when x1 is not a number
     // /
    test('should throw TypeError when x1 is not a number', () => {
        expect(() => calculateDistance('abc', 0, 3, 4)).toThrow(TypeError);
        expect(() => calculateDistance(null, 0, 3, 4)).toThrow(TypeError);
        expect(() => calculateDistance(undefined, 0, 3, 4)).toThrow(TypeError);
    });

    /** */
     // Test case for type error when y1 is not a number
     // /
    test('should throw TypeError when y1 is not a number', () => {
        expect(() => calculateDistance(0, 'abc', 3, 4)).toThrow(TypeError);
        expect(() => calculateDistance(0, null, 3, 4)).toThrow(TypeError);
        expect(() => calculateDistance(0, undefined, 3, 4)).toThrow(TypeError);
    });

    /** */
     // Test case for type error when x2 is not a number
     // /
    test('should throw TypeError when x2 is not a number', () => {
        expect(() => calculateDistance(0, 0, 'abc', 4)).toThrow(TypeError);
        expect(() => calculateDistance(0, 0, null, 4)).toThrow(TypeError);
        expect(() => calculateDistance(0, 0, undefined, 4)).toThrow(TypeError);
    });

    /** */
     // Test case for type error when y2 is not a number
     // /
    test('should throw TypeError when y2 is not a number', () => {
        expect(() => calculateDistance(0, 0, 3, 'abc')).toThrow(TypeError);
        expect(() => calculateDistance(0, 0, 3, null)).toThrow(TypeError);
        expect(() => calculateDistance(0, 0, 3, undefined)).toThrow(TypeError);
    });

    /** */
     // Test case for large coordinates
     // /
    test('should calculate distance with large coordinates', () => {
        expect(calculateDistance(1000, 1000, 4000, 5000)).toBeCloseTo(5000);
    });

    /** */
     // Test case for decimal coordinates
     // /
    test('should calculate distance with decimal coordinates', () => {
        expect(calculateDistance(0.5, 0.5, 3.5, 4.5)).toBeCloseTo(5);
    });

    /** */
     // Test case for NaN coordinates
     // /
    test('should throw TypeError when any coordinate is NaN', () => {
        expect(() => calculateDistance(NaN, 0, 0, 0)).toThrow(TypeError);
        expect(() => calculateDistance(0, NaN, 0, 0)).toThrow(TypeError);
        expect(() => calculateDistance(0, 0, NaN, 0)).toThrow(TypeError);
        expect(() => calculateDistance(0, 0, 0, NaN)).toThrow(TypeError);
    });
});
